### PR TITLE
Remove continue-on-error, improv GHA, add template

### LINKED ISSUE #406: Fix deploy action and JS moved content
* Links are changed now: `foreman.html` is now `foreman-el.html` - we need to update the JS that suggest where the new content is at least.
* Deploy action does not work - it did not update the site - need to fix.
* Deploy action should delete all files before copying new content but only in nightly or X.Y directories (not all that would delete all content)

### PR DESCRIPTION
**Pull request by lzap**
Turns out it was really bad idea to remove this, github reports everything as green. I was hoping for some yellow color. This is not very transparent, as a result we managed to merge some commit which broke links and now the deploy action which depends on link validation fails. That's why the site was never updated.

Fixes #406

### CODE REVIEW

**Comment by lzap** [comment]
I am looking into those failures, God this is a mess. :-)

**Comment by lzap** [comment]
So one problem was caused by @ehelms change of an anchor. that included "satellite" word and it was removed. This ultimately broke cross-doc links. Eric FYI please do not remove "satellite" from anchors (`[id=something]`), it will break links which are across guides.

We have a TODO item to add new anchors without satellite, then we need to wait one or two releases until it bubbles down to downstream and only after that we can do the change. It's not many, I can probably do this today:

```
guides/common/assembly_configuring-satellite-custom-server-certificate.adoc:4:[id="configuring-satellite-custom-server-certificate_{context}"]
guides/common/modules/proc_attaching-satellite-infrastructure-subscription.adoc:1:[id="attaching-satellite-infrastructure-subscription_{context}"]
guides/common/modules/proc_configuring-satellite-to-use-external-databases.adoc:1:[id="configuring-satellite-to-use-external-databases_{context}"]
guides/common/modules/proc_configuring-satellite-automatically-using-an-answer-file.adoc:1:[id="configuring-satellite-automatically-using-an-answer-file_{context}"]
guides/common/modules/proc_configuring-satellite-for-outgoing-emails.adoc:1:[id="configuring-satellite-for-outgoing-emails_{context}"]
guides/common/modules/proc_configuring-satellite-manually.adoc:1:[id="configuring-satellite-manually_{context}"]
guides/common/modules/proc_configuring-satellite-to-synchronize-content-with-a-local-cdn-server.adoc:1:[id="configuring-satellite-to-synchronize-content-with-a-local-cdn-server_{context}"]
guides/common/modules/proc_enabling-client-connections-to-satellite.adoc:1:[id="enabling-client-connections-to-satellite_{context}"]
guides/common/modules/proc_configuring-satellite-deployment-with-an-external-dhcp-server.adoc:1:[id="configuring-satellite-deployment-with-an-external-dhcp-server_{context}"]
guides/common/modules/proc_installing-the-satellite-server-packages.adoc:1:[id="installing-the-satellite-server-packages_{context}"]
guides/common/modules/proc_deploying-a-custom-ssl-certificate-to-satellite-server.adoc:1:[id="deploying-a-custom-ssl-certificate-to-satellite-server_{context}"]
guides/common/modules/proc_importing-a-subscription-manifest-into-satellite-server.adoc:1:[id="importing-a-subscription-manifest-into-satellite-server_{context}"]
guides/common/modules/proc_enabling-connections-from-capsule-to-satellite.adoc:1:[id="enabling-connections-from-capsule-to-satellite_{context}"]
guides/common/modules/proc_synchronizing-the-satellite-tools-repository.adoc:1:[id="synchronizing-the-satellite-tools-repository_{context}"]
guides/common/modules/proc_using-insights-with-satellite-server.adoc:1:[id='using-insights-with-satellite-server_{context}']
guides/common/modules/ref_satellite-ports-and-firewalls-requirements.adoc:1:[id="satellite-ports-and-firewalls-requirements_{context}"]
guides/common/modules/proc_enabling-the-satellite-tools-repository.adoc:1:[id="enabling-the-satellite-tools-repository_{context}"]
guides/common/modules/proc_reverting-satellite-to-download-content-from-red-hat-cdn.adoc:1:[id="reverting-satellite-to-download-content-from-red-hat-cdn_{context}"]
guides/common/modules/proc_listing-using-satellite-ansible-modules.adoc:1:[id="listing-using-satellite-ansible-modules_{context}"]
guides/common/modules/proc_installing-satellite-ansible-modules-via-rpm.adoc:1:[id="installing-satellite-ansible-modules-via-rpm_{context}"]
guides/common/modules/proc_registering-to-satellite-server.adoc:1:[id="registering-to-satellite-server_{context}"]
guides/common/assembly_configuring-satellite-with-an-http-proxy.adoc:1:[id="configuring-satellite-with-an-http-proxy_{context}"]
guides/common/assembly_installing-satellite-server-disconnected.adoc:1:[id="installing-satellite-server-disconnected"]
guides/common/assembly_installing-satellite-server-disconnected.adoc:28:[id="configuring-server_{context}"]
guides/common/assembly_preparing-environment-for-installation.adoc:1:[id="preparing-environment-for-satellite-installation"]
guides/common/assembly_using-satellite-ansible-content-collections.adoc:3:[id="using-satellite-ansible-content-collections_{context}"]
guides/common/assembly_installing-satellite-server-connected.adoc:1:[id="installing-satellite-server-connected"]
guides/common/assembly_installing-satellite-server-connected.adoc:69:[id="configuring-server_{context}"]
guides/doc-Administering_Red_Hat_Satellite/topics/Maintenance.adoc:119:[id='installing-and-updating-packages-on-satellite-server']
guides/doc-Administering_Red_Hat_Satellite/topics/assembly_backing-up-satellite-server-and-capsule-server.adoc:1:[id="backing-up-satellite-server-and-capsule-server"]
guides/doc-Administering_Red_Hat_Satellite/topics/configuring-idm-authentication-on-satellite-server.adoc:1:[id='configuring-idm-authentication-on-satellite-server_{context}']
guides/doc-Administering_Red_Hat_Satellite/topics/configuring-red-hat-satellite-to-use-ldap.adoc:1:[id='configuring-red-hat-satellite-to-use-ldap_{context}']
guides/doc-Administering_Red_Hat_Satellite/topics/enrolling-satellite-server-with-the-ad-server.adoc:1:[id='enrolling-satellite-server-with-the-ad-server_{context}']
guides/doc-Administering_Red_Hat_Satellite/topics/log-files-provided-by-satellite.adoc:1:[id='log-file-directories-provided-by-satellite_{context}']
guides/doc-Administering_Red_Hat_Satellite/topics/predefined-roles-available-in-satellite.adoc:1:[id='predefined-roles-available-in-satellite_{context}']
guides/doc-Administering_Red_Hat_Satellite/topics/proc_performing-a-full-backup-of-satellite-or-capsule.adoc:1:[id='performing-a-full-backup-of-satellite-or-capsule_{context}']
guides/doc-Administering_Red_Hat_Satellite/topics/satellite-settings.adoc:1:[id='satellite-settings_{context}']
guides/doc-Configuring_Load_Balancer/topics/Preparing_Satellite_Server_and_Capsule_Servers.adoc:1:[id='preparing-satellite-server-and-capsule-servers']
guides/doc-Configuring_Load_Balancer/topics/Overview.adoc:20:[id='satellite-load-balancing-solution-architecture']
guides/doc-Managing_Hosts/topics/proc_registering-a-host-to-satellite-using-the-bootstrap-script.adoc:1:[id="registering-a-host-to-satellite-using-the-bootstrap-script"]
guides/doc-Managing_Hosts/topics/proc_registering-a-host-to-satellite-using-the-bootstrap-script.adoc:130:[id="setting-permissions-for-the-bootstrap-script"]
guides/doc-Managing_Hosts/topics/proc_removing-a-host-from-satellite.adoc:1:[id="removing-a-host-from-satellite"]
guides/doc-Managing_Hosts/topics/proc_removing-a-host-from-satellite.adoc:21:[id="disassociating-a-virtual-machine"]
guides/doc-Managing_Hosts/topics/using_insights_with_satellite_hosts.adoc:1:[id="using-insights-with-satellite-hosts"]
guides/doc-Managing_Hosts/topics/proc_registering-an-atomic-host-to-satellite.adoc:1:[id="registering-an-atomic-host-to-satellite"]
guides/doc-Managing_Hosts/topics/proc_creating-a-host-in-satellite.adoc:1:[id="creating-a-host-in-satellite"]
guides/doc-Managing_Hosts/topics/proc_registering-a-host-to-satellite-manually.adoc:1:[id="registering-a-host-to-satellite"]
guides/doc-Managing_Hosts/topics/con_overview-of-hosts-in-satellite.adoc:1:[id="overview-of-hosts-in-satellite"]
```

**Comment by lzap** [comment]
The second problem was @ekohl bumping release to 2.5, I understand the motivation, we have branched off the core already, however documentation is catching up and I doubt people are writing documentation that is only valid for 2.5. We will branch off but definitely later, it makes little sense to branch off now - RH docs team is pushing a lot of content for 6.9/2.4 right now.

The error is that the variable is used in yum repository examples and https://yum.theforeman.org/client/2.5/ is 404. For now I am decreasing versions back to 2.4 and I would suggest to bump it up only after we branch off in few weeks. I want to get HTML link validation passing again, we can implement some kind of `ProjectVersionLatestStable` attribute to workaround the issue later on.

**Comment by lzap** [comment]
The last issue is straight-up 404 on our portal for a link that used to work: https://access.redhat.com/documentation/en-us/red_hat_insights/2020-04/html/client_configuration_guide_for_red_hat_insights/index

Google still finds it but it's 404.

**Comment by lzap** [comment]
Two more changes:

* I am improving our HTML validator by installing "retryer" python app, it retries when a link is broken. This should improve those failures which are coming from our unreliable portal which timeouts quite often.
* I moved PDF generation to be called after publish action, we never had a failure for PDF builds, we don't publish them and it's slow.

**Comment by lzap** [comment]
You know what, I am changing my mind with moving PDF generation after deploy. HTML link validation takes about 3 minutes and generating those PDFs also takes 3 minutes. Since GHA runs everything in parallel, we are good. It's handy to be able to check PDFs.

I am done, this is now ready to be merged.

**Comment by lzap** [comment]
Feel free to use my new cherry-picking script, it's safe and interactive: https://github.com/lzap/bin-public/blob/master/git-xcp

Just drop it to your `$PATH` and run it.

**Review by melcorr — APPROVED** [review]
LGTM

File: guides/common/assembly_preparing-environment-for-installation.adoc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-[id="preparing-environment-for-installation"]
+[id="preparing-environment-for-satellite-installation"]
```
Comment (ehelms) [inline guides/common/assembly_preparing-environment-for-installation.adoc:1]: This is confusing. Shouldn't the anchor reflect the same name as the file? And as long as all references within the branch use the same id that everything should work?

**Pull request merged**